pkg/controller/certificates/issuing/internal/secret.go

```diff
@@ -59,7 +59,9 @@ type SecretsManager struct {
 
 // SecretData is a structure wrapping private key, Certificate and CA data
 type SecretData struct {
-	PrivateKey, Certificate, CA []byte
+	PrivateKey, Certificate, CA         []byte
+	CertificateName                     string
+	IssuerName, IssuerKind, IssuerGroup string
```

Comment:
We now explicitly pass the CertificateName, IssuerName, IssuerKind and IssuerGroup values to setValues instead reading these values from the cmapi.Certificate resource, since cmapi.Certificate might have been updated since the secret was created.